### DISCUSSION TITLE
Using Existing Actions In infolist "anonymous" actions

### DISCUSSION THREAD

**Discussion by bomshteyn** [filamentphp/filament#7535] [Help]
I am trying to add an Edit Action inside the Infolist like so:

```PHP
Actions::make([
    EditAction::make()
])
->fullWidth(),
```

But am getting an error 
```
Filament\Infolists\Components\Actions::Filament\Infolists\Components\{closure}(): Argument #1 ($action) must be of type Filament\Infolists\Components\Actions\Action, Filament\Actions\EditAction given
```

Is this by design or an Issue that should be reported?

Also if i already have this action in the header actions is there a way to trigger that action without having the full duplicate action?

**Comment by danharrin** [comment]
EditAction is a different type of action and can't be used here, so you just need to copy the configuration from there onto your own `Filament\Infolists\Components\Actions\Action`

**Reply by bomshteyn** [reply]
Can't it be made not as strict so as so to allow other Action types to avoid duplicating the code?

**Reply by danharrin** [reply]
No, there is a reason why different types of action use different classes, since they have different features
